### PR TITLE
Feature/40 implement analyzer tests

### PR DESCRIPTION
**Pull request by Gianluca005**
This pull request refactors and updates the unit tests related to license analysis, focusing on improved modularity, clarity, and use of modern testing frameworks. The key changes include replacing legacy `unittest`-based tests with more targeted and maintainable `pytest`-based tests, introducing new test modules for core functionalities, and fixing a minor bug in the license comparator logic.

**Test suite modernization and reorganization:**

* Removed legacy `unittest`-based test modules: `test_license_analyzer.py` and `test_license_compatibility_analyzer.py` have been deleted to reduce redundancy and improve maintainability. [[1]](diffhunk://#diff-6b785edae83bd09e5281cabc29a64fb2800e9df7c31d207aaa9d59b49080a590L1-L59) [[2]](diffhunk://#diff-a5a57e2fd6b4c7c970518bb6709699dee57d13ec34350ec13345fd24d92a05dcL1-L361)
* Added new `pytest`-based test modules for improved coverage and clarity:
  - [`test_license_comparator.py`](diffhunk://#diff-c637aabf211d381419c0f037e398fb606f437c6ec1dc3032f0e522f6d1f37509R1-R91): Thoroughly tests `LicenseComparator`, including edge cases and error handling.
  - [`test_license_name_normalizer.py`](diffhunk://#diff-7e23d864194e906f0ddbbf34c3e8920da94be717fb68fc7e294cc529c3047bffR1-R31): Adds focused tests for the `license_name_normalizer` module, including normalization logic and error scenarios.

**Bug fix:**

* Fixed a bug in `compare_license_trees` in `license_comparator.py` where items were incorrectly appended to the `doubts` list (now correctly appends as a tuple).

### CODE REVIEW

File: test/unittest/analyzer/test_license_comparator.py [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+
+    def test_compare_license_trees_missing_in_scan(self, mock_scan_engine, pypi_metadata_factory):
+        """Test error handling when scan returns empty list."""
+        meta = pypi_metadata_factory("PkgA", "MIT")
+        mock_scan_engine.scan_for_license.return_value = []
+        
+        comparator = LicenseComparator([meta], mock_scan_engine)
+        
+        with pytest.raises(IndexError):
+            comparator.compare_license_trees()
```
[the remark this answers is not in the document]
Comment (kbublij2) [reply test/unittest/analyzer/test_license_comparator.py]: Upstream source of `IndexError` is now fixed, this change is no more necessary

**Comment by kbublij2** [comment]
Pylint cleanup will have to happen at an appriopriate time, all else LGTM
